src/on_demand/aca_on_demand_engine.cpp

```diff
@@ -107,6 +139,7 @@ void ACA_On_Demand_Engine::parse_packet(uint32_t in_port, void *packet)
   string ip_src, ip_dest;
   int port_src, port_dest;
   Protocol _protocol = Protocol::TCP;
+  OperationStatus on_demand_reply = OperationStatus::FAILURE;
```

Comment:
don't need to set to OperationStatus::FAILURE here because you will be setting later. This will give compile a chance to catch coding mistakes (if any).

Reply:
the default setting is removed.